lib/matplotlib/colors.py

```diff
@@ -1225,6 +1254,875 @@ def reversed(self, name=None):
+          is displayed
+        - If 'ignore' the variates are not clipped, but instead assigned the
+          'outside' color
+        - If 'circleignore' a circular mask is applied, but the data is not clipped
+
+    origin: (float, float)
+        The relative origin of the colormap. Typically (0, 0), for colormaps
+        that are linear on both axis, and (.5, .5) for circular colormaps.
+        Used when getting 1D colormaps from 2D colormaps.
+
+    name : str, optional
+        The name of the colormap.
+    """
+
+    def __init__(self, patch, N=256, shape='square', origin=(0, 0),
+                 name='segmented bivariate colormap'):
+        self.patch = patch
+        super().__init__(N, N, shape, origin, name=name)
+
+    def _init(self):
+        s = self.patch.shape
+        _patch = np.empty((s[0], s[1], 4))
+        _patch[:, :, :3] = self.patch
+        _patch[:, :, 3] = 1
+        transform = mpl.transforms.Affine2D().translate(-0.5, -0.5)\
+                                .scale(self.N / (s[0] - 1), self.N / (s[0] - 1))
+        self._lut = np.empty((self.N, self.N, 4))
+
+        _image.resample(_patch, self._lut, transform, _image.BILINEAR,
+                        resample=False, alpha=1)
+        self._isinit = True
+
+
+class BivarColormapFromImage(BivarColormap):
+    """
+    BivarColormap object generated by supersampling a regular grid
+
+    Parameters
+    ----------
+    lut : nparray of shape (N, M, 3) or (N, M, 4)
+        The look-up-table
+    shape: str 'square' or 'circle' or 'ignore' or 'circleignore'
+
+        - If 'square' each variate is clipped to [0,1] independently
+        - If 'circle' the variates are clipped radially to the center
+          of the colormap, and a circular mask is applied when the colormap
+          is displayed
+        - If 'ignore' the variates are not clipped, but instead assigned the
+          'outside' color
+        - If 'circleignore' a circular mask is applied, but the data is not clipped
+
+    origin: (float, float)
+        The relative origin of the colormap. Typically (0, 0), for colormaps
+        that are linear on both axis, and (.5, .5) for circular colormaps.
+        Used when getting 1D colormaps from 2D colormaps.
+    name : str, optional
+        The name of the colormap.
+
+    """
+
+    def __init__(self, lut, shape='square', origin=(0, 0), name='from image'):
+        # We can allow for a PIL.Image as unput in the following way, but importing
+        # matplotlib.image.pil_to_array() results in a circular import
+        # For now, this function only accepts numpy arrays.
+        # i.e.:
+        # if isinstance(Image, lut):
+        #    lut = image.pil_to_array(lut)
+        lut = np.array(lut, copy=True)
+        if lut.ndim != 3 or lut.shape[2] not in (3, 4):
+            raise ValueError("The lut must be an array of shape (n, m, 3) or (n, m, 4)",
+                             " or a PIL.image encoded as RGB or RGBA")
```

Comment:
Probably this should use `_api.check_shape` too, but it doesn't appear to accept a dimension with two options.

Reply:
I think there is some similar logic for accepting RGB images in imshow, so if `_api.check_shape` was adapted to check multiple cases, it could possibly be used there as well.

I'm leaving this as it is for now.